### PR TITLE
Added cast for FixedSizeBinary to (Large)Binary

### PR DESCRIPTION
**Pull request by ritchie46**

### CODE REVIEW

File: src/array/fixed_size_binary/mod.rs [hunk 1]
```diff
@@ -87,6 +87,12 @@ impl FixedSizeBinaryArray {
 
 // must use
 impl FixedSizeBinaryArray {
+    /// Retrieve the size of the binary elements
+    /// in this ['FixedSizeBinaryArray']
+    pub fn fixed_size(&self) -> usize {
```
Comment (jorgecarleitao) [inline src/array/fixed_size_binary/mod.rs:92]:
```suggestion
    pub fn size(&self) -> usize {
```

(to be aligned with FixedSizedListArray)

File: src/compute/cast/binary_to.rs [hunk 2]
```diff
@@ -118,3 +118,44 @@ pub(super) fn binary_to_dictionary_dyn<O: Offset, K: DictionaryKey>(
...
+    unsafe { Offsets::new_unchecked(offsets) }
+}
+
+/// Conversion of large-binary
+pub fn fixed_size_binary_to_large_binary(
+    from: &FixedSizeBinaryArray,
+    to_data_type: DataType,
+) -> BinaryArray<i64> {
+    let values = from.values().clone();
+    let offsets = fixed_size_to_offsets(values.len(), from.fixed_size());
```
Comment (jorgecarleitao) [inline src/compute/cast/binary_to.rs:139]:
```suggestion
    let offsets = fixed_size_to_offsets(values.len(), from.size());
```

File: src/compute/cast/binary_to.rs [hunk 3]
```diff
@@ -118,3 +118,44 @@ pub(super) fn binary_to_dictionary_dyn<O: Offset, K: DictionaryKey>(
...
+    )
+}
+
+/// Conversion of binary
+pub fn fixed_size_binary_to_binary(
+    from: &FixedSizeBinaryArray,
+    to_data_type: DataType,
+) -> BinaryArray<i32> {
+    let values = from.values().clone();
+    let offsets = fixed_size_to_offsets(values.len(), from.fixed_size());
```
Comment (jorgecarleitao) [inline src/compute/cast/binary_to.rs:154]:
```suggestion
    let offsets = fixed_size_to_offsets(values.len(), from.size());
```

File: src/compute/cast/binary_to.rs [hunk 4]
```diff
@@ -118,3 +118,44 @@ pub(super) fn binary_to_dictionary_dyn<O: Offset, K: DictionaryKey>(
...
+        .map(|v| O::from_usize(v).unwrap())
+        .collect();
+    // Safety
+    // * every element is `>= 0`
+    // * element at position `i` is >= than element at position `i-1`.
+    unsafe { Offsets::new_unchecked(offsets) }
+}
+
+/// Conversion of large-binary
+pub fn fixed_size_binary_to_large_binary(
```
Comment (jorgecarleitao) [inline src/compute/cast/binary_to.rs:134]: Maybe refactor this into a generic with `<O: Offset>` to cover both cases at once?

**Review by jorgecarleitao — APPROVED** [review]
Thanks a lot, @ritchie46 ! Looks great; left minor comments / suggestions but feel free to merge it as is.

File: src/compute/cast/binary_to.rs (hunk 4 above)
Comment (ritchie46) [reply src/compute/cast/binary_to.rs:134]: Right, good one. That I didn't see this.

**Pull request merged**